prow/resultstore/writer/writer.go

```diff
@@ -70,6 +72,24 @@ type writer struct {
 	finalized   bool
 }
 
+// persistentError returns whether the error status code is not
+// retryable per the ResultStore maintainers. Any codes not
+// listed here should be retried with exponential backoff.
+func persistentError(err error) bool {
+	status, _ := status.FromError(err)
+	switch status.Code() {
+	case codes.AlreadyExists:
+	case codes.NotFound:
+	case codes.InvalidArgument:
+	case codes.FailedPrecondition:
+	case codes.Unimplemented:
+	case codes.PermissionDenied:
+	default:
+		return false
+	}
+	return true
```

Comment:
This code is tricky. It's saying that if we see a match, we go to line 90 (it is a persistent error). But I would find it much easier to read if you just put in the explicit `fallthrough` in each case, and then have the last matching case return `true`, and delete the separate `default` case. Then in the outer block return `false`. This way the return value and the cases that match are not separated from each other.

Reply:
Done